### ISSUE TITLE
Update exon selection offset to consider cases where breakpoint occurs before or after a transcript

### ISSUE THREAD

**Issue by jarbesfeld** [GenomicMedLab/cool-seq-tool#380]
### Describe the bug

There are cases where a breakpoint can occur before or after a transcript (i.e. in the untranslated region, or UTR). We should check for these cases when selecting the adjacent exon.

### Steps to reproduce

```shell
resp = test_egc_mapper._get_adjacent_exon(
        tx_exons_genomic_coords=nm_152263_exons_genomic_coords,
        start=154192102,
        strand=Strand.POSITIVE,
    )
```

### Expected behavior

This should output 1, since this occurs two bases before the transcription start site of exon 1

### Current behavior

This returns exon 10, as it iterates through the whole exon list since this breakpoint occurs outside of the transcript range

### Branch, commit, and/or version

Main branch

### Environment details

Mac M1

### Contribution

Yes, I can create a PR for this fix.

**Comment by korikuzma** [comment]
@jarbesfeld is this similar to #361?

**Comment by jarbesfeld** [comment]
@korikuzma Yes this includes a fix to the logic described in #361. I realized when looking at the GENIE data that there can be breakpoint before or after the transcript (e.g. `Promoter of AOC4P(+):3Kb from tx start`), so I thought it would be good to add this fix
